src/coreComponents/unitTests/meshTests/testCornerPointMeshImport.cpp

```diff
@@ -0,0 +1,265 @@
+                     string const & inputStringRegion,
+                     string const & meshBodyName,
+                     string const & propertyToTest )
+{
+  conduit::Node node;
+  Group root( "root", node );
+  MeshManager meshManager( "mesh", &root );
+
+  // Load the mesh
+  xmlWrapper::xmlDocument xmlDocument;
+  xmlDocument.load_buffer( inputStringMesh.c_str(), inputStringMesh.size() );
+
+  xmlWrapper::xmlNode xmlMeshNode = xmlDocument.child( "Mesh" );
+  meshManager.processInputFileRecursive( xmlMeshNode );
+  meshManager.postProcessInputRecursive();
+  // we stop here for nonconfornming open vtk files
+  // file-file comparison
+  // write a function to parse and compare the generated files
+
+  // Create the domain and generate the Mesh
+  DomainPartition domain( "domain", &root );
+  meshManager.generateMeshes( domain );
+
+  // nonconforming stops here
+
+  MeshBody & meshBody = domain.getMeshBody( meshBodyName );
+  MeshLevel & meshLevel = meshBody.getMeshLevel( 0 );
+  NodeManager & nodeManager = meshLevel.getNodeManager();
+  FaceManager const & faceManager = meshLevel.getFaceManager();
+  ElementRegionManager & elemManager = meshLevel.getElemManager();
+
+  // Create the ElementRegions
+  xmlDocument.load_buffer( inputStringRegion.c_str(), inputStringRegion.size() );
+
+  xmlWrapper::xmlNode xmlRegionNode = xmlDocument.child( "ElementRegions" );
+  elemManager.processInputFileRecursive( xmlRegionNode );
+  elemManager.postProcessInputRecursive();
+
+  CellBlockManager & cellBlockManager = meshBody.getGroup< CellBlockManager >( keys::cellManager );
+  nodeManager.setGeometricalRelations( cellBlockManager, elemManager );
+
+  elemManager.generateMesh( cellBlockManager );
+
+  if( !propertyToTest.empty() )
+  {
+    elemManager.forElementSubRegions< ElementSubRegionBase >( [&]( ElementSubRegionBase & subRegion )
+    {
+      subRegion.registerWrapper< array1d< real64 > >( propertyToTest );
+    } );
+  }
+
+  // Trigger import of the field
+  meshManager.importFields( domain );
+
+  // Check if the computed center match with the imported center
+  if( !propertyToTest.empty() )
+  {
+    real64 const expectedSumPoreVolume = 1062756.27012;
+    real64 sumPoreVolume = 0.0;
+    real64 const eps = meshBody.getGlobalLengthScale() * 1e-8;
+    elemManager.forElementSubRegions< ElementSubRegionBase >( [&]( ElementSubRegionBase & subRegion )
+    {
+      subRegion.calculateElementGeometricQuantities( nodeManager, faceManager );
+      arrayView1d< real64 const > const elemVolume = subRegion.getElementVolume();
+      arrayView1d< real64 const > const porosityProperty = subRegion.getReference< array1d< real64 > >( propertyToTest );
+
+      for( localIndex ei = 0; ei < subRegion.size(); ++ei )
+      {
+        sumPoreVolume += elemVolume[ei] * porosityProperty[ei];
+      }
+    } );
+    EXPECT_LE( LvArray::math::abs( sumPoreVolume - expectedSumPoreVolume ), eps );
+  }
+}
+
+string fileToString( const string filePath )
```

Comment:
I'd be surprised that we do not have already this kind of utilities. @francoishamon ?